code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -209,6 +209,36 @@
 		return 0
 	return ..()
 
+//////////////////////////////////////////////
+//                                          //
+//               TIME AGENT                 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/time_agent
+	name = "time agent anomaly"
+	role_category = /datum/role/time_agent
+	required_candidates = 1
+	weight = 4
+	cost = 10
+	requirements = list(70, 60, 50, 40, 30, 20, 10, 10, 10, 10)
+	logo = "time-logo"
+
+/datum/dynamic_ruleset/midround/from_ghosts/time_agent/acceptable(var/population=0,var/threat=0)
+	var/player_count = mode.living_players.len
+	var/antag_count = mode.living_antags.len
+	var/max_traitors = round(player_count / 10) + 1
```

Comment:
this is fine but probably not necessary - the purpose of this is to regulate overall traitor numbers for modes like traitor and vampire that make a lot of antagonists, but time agent only makes one

ideally the dynamic mode already takes into account number of living antagonists and won't roll a new ruleset while there are several alive except in a very low chance